backend/api/services/config_service.js

```diff
@@ -0,0 +1,92 @@
+var config_api = require('../../config.js');
+var path = require('path');
+var fs = require('fs-extra');
+var {logger, logger_transports} = require('./logger_service');
+
+var validDownloadingAgents = [
+  'aria2c',
+  'avconv',
+  'axel',
+  'curl',
+  'ffmpeg',
+  'httpie',
+  'wget'
+];
+
+
+var backendPort = null;
+var usingEncryption = null;
+var audioFolderPath = null;
+var videoFolderPath = null;
+var downloadOnlyMode = null;
+var useDefaultDownloadingAgent = null;
+var customDownloadingAgent = null;
+var allowSubscriptions = null;
+var subscriptionsCheckInterval = null;
+
+// other needed values
+var options = null; // encryption options
+var url_domain = null;
+
+function loadConfigValues() {
+  url = !debugMode ? config_api.getConfigItem('ytdl_url') : 'http://localhost:4200';
```

Comment:
debugMode might not be imported properly here

Reply:
Don't think so, here's the logic for it:

`let debugMode = process.env.YTDL_MODE === 'debug';`

I see that's in application.js, but I don't mind it being in the config service since other modules will likely need it. Otherwise we can be messy and include that line at the top of any files that need that variable.